src/components/App.tsx

```diff
@@ -195,7 +195,13 @@ const App = () => {
           pathname === '/app' ||
           pathname === '/app/storages'
         ) {
-          if (localSpaces.length > 0) {
+          // read location from local storage and set initial page from it if it exists
+          const previousPathname = localStorage.getItem(
+            'lastOpenedPagePathname'
+          )
+          if (previousPathname) {
+            push(previousPathname)
```

Comment:
Technically this pathname could be invalid, we could check for empty string etc.
If user goes to some pathname which is not saved (but all should be), and for example deletes storage, it might happen that on load it tries to load some 404 pathname and thus giving problems, any ideas how to check if this pathname is at least somewhat correct and if not default to valid options? (maybe parse the pathname contents, check if storage exists, if document exists)

This is especially important in team cloud spaces, as document could be removed from other teammates?